### DISCUSSION TITLE
Lomse xml and color attribute

### DISCUSSION THREAD

**Discussion by denisgottardello** [lenmus/lomse#401] [General]
Hi, is color attribute supported in note element? If yes can someone share an xml example?

**Comment by cecilios** [comment]
In MusixXML the color is controlled by the `color` attribute in the `<note>`, `<notehead>` and other elements:
     https://www.w3.org/2021/06/musicxml40/musicxml-reference/data-types/color/

For example, to add color just to a notehead:
```
<notehead color="#FF0000">normal</notehead>
```
Or to add color to all (notehead, stem and flag) or to a rest:

```
  <note color="#FF0000">
    <rest/>
```

Currently the MusicXML importer in Lomse ignores all `%print-style` related attributes, such as color. But it's probably easy to modify the code to take color into account for some elements already supporting color, such as notes.

**Comment by denisgottardello** [comment]
Many thanks for your reply but it seems not working.

using color attribute in note element like this
      <note *color="#FF0000"*>
        <pitch>
          <step>C</step>
          <octave>4</octave>
        </pitch>
        <duration>4</duration>
        <type>quarter</type>
      </note>

or
      <note>
        <pitch>
          <step>C</step>
          <octave>4</octave>
        </pitch>
        <duration>4</duration>
        <type>quarter</type>
        *<notehead color="#FF0000">normal</notehead>*
      </note>
I still see the node in black

In data lunedì 21 ottobre 2024 16:32:13 CEST, Cecilio Salmeron ha scritto:
> In MusixXML the color is controlled by the `color` in the `<note>`,
> `<notehead>` and other elements:
> https://www.w3.org/2021/06/musicxml40/musicxml-reference/data-types/color/
> 
> For example, to add color just to a notehead:
> ```
> <notehead color="#FF0000">normal</notehead>
> ```
> Or to add color to all (notehead, stem and flag) or to a rest:
> 
> ```
>   <note color="#FF0000">
>     <rest/>
> ```
> 
> Currently the MusicXML importer in Lomse ignores all `%print-style` related
> attributes, such as color. But it's probably easy to modify the code to
> take color into account for some elements already supporting color, such as
> notes.
-- 
+39.347.4070897
http://www.labcsp.com[1] 
http://www.denisgottardello.it[2] 
GMT+1
Skype: mrdebug 

--------
[1] http://www.labcsp.com
[2] http://www.denisgottardello.it

**Comment by cecilios** [comment]
> I still see the node in black

I already told you. Please read the last paragraph of my previous answer.

If this is important for your application, please open an issue

**Comment by denisgottardello** [comment]
Oh it is not supported. Is there another way to highlight a particular note right now?

In data martedì 22 ottobre 2024 10:23:46 CEST, Cecilio Salmeron ha scritto:
> > I still see the node in black
> 
> I already told you. Please read the last paragraph of my previous answer.
> 
> If this is important for your application, please open an issue
-- 
+39.347.4070897
http://www.labcsp.com[1] 
http://www.denisgottardello.it[2] 
GMT+1
Skype: mrdebug 

--------
[1] http://www.labcsp.com
[2] http://www.denisgottardello.it

**Comment by cecilios** [comment]
> Is there another way to highlight a particular note right now?

No, but if you need this, please open an issue and I will fix it along before the weekend. It is not complex.

**Comment by denisgottardello** [comment]
Many thanks. I just have upgraded my software using the note color attribute.
